pkg/azureclients/vmssclient/azure_vmssclient.go

```diff
@@ -288,39 +288,30 @@ func (c *Client) CreateOrUpdateAsync(ctx context.Context, resourceGroupName stri
 
 // WaitForCreateOrUpdateResult waits for the response of the create or update request
 func (c *Client) WaitForCreateOrUpdateResult(ctx context.Context, future *azure.Future, resourceGroupName string) (*http.Response, error) {
-	mc := metrics.NewMetricContext("vmss", "wait_for_create_or_update_result", resourceGroupName, c.subscriptionID, "")
-	res, err := c.armClient.WaitForAsyncOperationResult(ctx, future, "VMSSWaitForCreateOrUpdateResult")
-	_ = mc.Observe(err)
-	return res, err
+	return c.WaitForAsyncOperationResult(ctx, future, resourceGroupName, "wait_for_create_or_update_result", "VMSSWaitForCreateOrUpdateResult")
```

Comment:
Cleaned up the duplication and have everything call into `WaitForAsyncOperationResult`

Reply:
much cleaner now, thanks for the improvement